Extensions/Lighting/lightruntimeobject-pixi-renderer.js

```diff
@@ -0,0 +1,254 @@
+gdjs.LightRuntimeObjectPixiRenderer = function (runtimeObject, runtimeScene) {
+  this._object = runtimeObject;
+  this._manager = runtimeObject.getObstaclesManager();
+  this._geometry = new PIXI.Geometry();
+  this._radius = runtimeObject.getRadius();
+  this._color = runtimeObject.getColor().map(function (item) {
```

Comment:
In most of the cases map could be easily avoided :) If it's better to use verbose way then we should do that. Also using declarative methods was the reason for issues with rendering :D

Reply:
I heavily favor them in the IDE, because in the app it's more important to have a very clear code and you can convey with them the transformation on "data", which is flowing from one function to the other.

In the game engine, the approach is still about keeping things clear of course, but we have to ensure a maximum performance as possible and for this try to allocate "statically" the objects as much as possible, reuse existing variables/arrays, minimize copies of data. Which often means relying to "good old" manual loops, and even avoid creating objects if two numbers, or a few variables can do the trick :) 
You can see at various place in the game engine that instead of passing a new object (for example, the result of raycastTest), it's a static object that is returned (the same object which is always reused).

Reply:
Unrelated stuff, I've seen some code in JsExtension, `.setType('colour')`, should I use that instead of string for color?

Reply:
Could be a good idea for consistency. This color is stored as a string, you'll need to do transform it, search for `hexToRGBColor` in the codebase and make a static method for it if needed :)